src/wireviz/DataClasses.py

```diff
@@ -214,6 +214,20 @@ def __post_init__(self) -> None:
         else:
             pass  # gauge not specified
 
+        if isinstance(self.length, str):  # length and unit specified
+            try:
+                l, u = self.length.split(' ')
+                l = float(l)
+            except Exception:
+                raise Exception(f'{self.length} - Length must be a number, or number and unit separated by a space')
+            self.length = l
+            self.length_unit = u
+        elif self.length is not None:  # length specified, assume m
+            if self.gauge_unit is None:
+                self.length_unit = 'm'
+        else:
+            pass  # length not specified
+
         self.connections = []
 
         if self.length_unit is None: #Default wire length units to meters if left undeclared
```

Comment:
This `if` and the belonging next line (github don't allow me to comment code too far off your changes) is not needed twice. In the new code above it is done only when needed.